File: app/src/main/java/com/yourapp/ui/MainActivity.java
package com.yourapp.ui;

import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.appcompat.app.AppCompatActivity;

import com.yourapp.R;
import com.yourapp.bridge.WebAppInterface;

public class MainActivity extends AppCompatActivity {

    private WebView webView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        webView = findViewById(R.id.webview);

        WebSettings ws = webView.getSettings();
        ws.setJavaScriptEnabled(true);
        ws.setDomStorageEnabled(true);

        ws.setLoadWithOverviewMode(true);
        ws.setUseWideViewPort(true);

        webView.setWebViewClient(new WebViewClient());

        // JS → Android bridge: JS side se "Android.xxx()" call yahin bind hota hai
        webView.addJavascriptInterface(new WebAppInterface(this), "Android");

        // CLOUDFLARE UI: yahi line decide karti hai UI kahan se load hoga
        webView.loadUrl("https://kyapata.sauravjha.workers.dev/");
        // Agar future me local asset use karna ho:
        // webView.loadUrl("file:///android_asset/index.html");
    }

    @Override
    public void onBackPressed() {
        if (webView != null && webView.canGoBack()) {
            webView.goBack();
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/com/example/raone/app/src/main/java/com/example/raone/ui/MainActivity.kt
package com.example.raone.ui

import android.os.Bundle
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import com.example.raone.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val webView = findViewById<WebView>(R.id.webview)

        val settings: WebSettings = webView.settings
        settings.javaScriptEnabled = true
        settings.domStorageEnabled = true

        webView.webViewClient = WebViewClient()
        webView.loadUrl("https://kyapata.sauravjha.workers.dev/")
    }
}
